From 50049d7e482a4e391335c35f3fac8122038bcff3 Mon Sep 17 00:00:00 2001
From: Xilin Wu <sophon@radxa.com>
Date: Wed, 23 Jul 2025 22:45:46 +0800
Subject: [PATCH 43/49] arm64: dts: qcom: qcs6490-radxa-dragon-q6a: Fix
 pcie0_clkreq_n and add missing reference to pcie1_clkreq_n

This allows PCIe ASPM to work properly.

Enabling ASPM of the RTL8111K requires this command:

`echo 1 | sudo tee /sys/devices/platform/soc@0/1c00000.pcie/pci0000:00/0000:00:00.0/0000:01:00.0/link/*`

Signed-off-by: Xilin Wu <sophon@radxa.com>
---
 .../arm64/boot/dts/qcom/qcs6490-radxa-dragon-q6a.dts | 12 +++++++++++-
 1 file changed, 11 insertions(+), 1 deletion(-)

diff --git a/src/arch/arm64/boot/dts/qcom/qcs6490-radxa-dragon-q6a.dts b/src/arch/arm64/boot/dts/qcom/qcs6490-radxa-dragon-q6a.dts
index 8f51f9490721..62890bd4a827 100644
--- a/src/arch/arm64/boot/dts/qcom/qcs6490-radxa-dragon-q6a.dts
+++ b/src/arch/arm64/boot/dts/qcom/qcs6490-radxa-dragon-q6a.dts
@@ -731,7 +731,7 @@ &pcie1 {
 	perst-gpios = <&tlmm 2 GPIO_ACTIVE_LOW>;
 	wake-gpios = <&tlmm 3 GPIO_ACTIVE_HIGH>;
 
-	pinctrl-0 = <&pcie1_reset_n>, <&pcie1_wake_n>;
+	pinctrl-0 = <&pcie1_reset_n>, <&pcie1_wake_n>, <&pcie1_clkreq_n>;
 	pinctrl-names = "default";
 
 	iommu-map = <0x0 &apps_smmu 0x1c80 0x1>,
@@ -1328,6 +1328,16 @@ &dp_hot_plug_det {
 	bias-disable;
 };
 
+&pcie0_clkreq_n {
+	bias-pull-up;
+	drive-strength = <2>;
+};
+
+&pcie1_clkreq_n {
+	bias-pull-up;
+	drive-strength = <2>;
+};
+
 &sdc2_clk {
 	bias-disable;
 	drive-strength = <16>;
-- 
2.50.1

